reddit-background

```diff
@@ -139,10 +147,14 @@ class Desktop(object):
 
         random.shuffle(image_urls)
 
-        for image_url in image_urls:
+        log('Number of images to download: {0}'.format(NUMBER_OF_IMAGES))
+        number_downloaded = 0
+        while (number_downloaded < NUMBER_OF_IMAGES):
             try:
-                return _download_image(image_url)
+                _download_image(image_urls[number_downloaded])
+                number_downloaded += 1
             except URLOpenError:
+                number_downloaded += 1
```

Comment:
This error indicates that we couldn't download it, so we should probably not count it in the tally. We should treat it as a skip, and move on to the next one.

With that in mind, the `number_downloaded += 1` statement could go in the `else` clause of the `try/except`